core/terraform/network/outputs.tf

```diff
@@ -73,9 +73,9 @@ output "file_core_dns_zone_id" {
 }
 
 output "queue_core_dns_zone_id" {
-  value = azurerm_private_dns_zone.private_dns_zones["privatelink.queue.core.windows.net"].id
+  value = azurerm_private_dns_zone.private_dns_zones[module.cloud_settings.private_links["privatelink.queue.core.windows.net"]].id
```

Comment:
So the name of the TF resource depends on the cloud it is deployed on. 
I think things might be related to my other comments...